soc/board_specific_workflows/test_general.py

```diff
@@ -0,0 +1,165 @@
+        self.build_name = 'test_build_name'
+        self.soc.build_name = self.build_name
+
+        # Autospec the Builder.
+        self.builder = mock.create_autospec(builder.Builder, instance=True)
+        self.gateware_dir = 'test_gateware_dir/'
+        self.builder.gateware_dir = self.gateware_dir
+
+        self.builder_constructor = mock.create_autospec(
+            builder.Builder, return_value=self.builder)
+        # Can't easily autospec the soc_constructor -- interface inconsistent.
+        self.soc_constructor = mock.MagicMock(return_value=self.soc)
+
+    def simple_init(self):
+        """Returns a GeneralSoCWorkflow with reasonable testing parameters."""
+        return general.GeneralSoCWorkflow(self.args, self.soc_constructor,
+                                          self.builder_constructor)
+
+    def test_init(self):
+        """Tests functionality of __init__ method."""
+        flow = self.simple_init()
+
+        self.assertEqual(flow.args, self.args)
+        self.assertEqual(flow.soc_constructor, self.soc_constructor)
+        self.assertEqual(flow.builder_constructor, self.builder_constructor)
+
+    def test_make_soc(self):
+        """Tests functionality of the make_soc method."""
+        in_kwargs = {'abcd': 'efgh'}
+        soc = self.simple_init().make_soc(**in_kwargs)
+        soc_kwargs = self.soc_constructor.call_args.kwargs
+
+        self.soc_constructor.assert_called_once()
+        self.assertEqual(soc, self.soc)
+        self.assertNotIn('toolchain', soc_kwargs)
+        self.assertEqual(in_kwargs['abcd'], soc_kwargs['abcd'])
+        self.assertEqual(self.args.with_ethernet, soc_kwargs['with_ethernet'])
+        self.assertEqual(self.args.with_etherbone,
+                         soc_kwargs['with_etherbone'])
+        self.assertEqual(self.args.with_mapped_flash,
+                         soc_kwargs['with_mapped_flash'])
+
+    @mock.patch('litex.soc.integration.soc_core.soc_core_argdict')
+    def test_make_soc_argdict_call(self, mock_soc_core_argdict):
+        """Tests soc_core_argdict is called when making the SoC."""
+        argdict_kwargs = {'abcd': 'efgh'}
+        mock_soc_core_argdict.return_value = argdict_kwargs
+        self.simple_init().make_soc()
+        soc_kwargs = self.soc_constructor.call_args.kwargs
+
+        mock_soc_core_argdict.assert_called_once()
+        self.assertEqual(argdict_kwargs['abcd'], soc_kwargs['abcd'])
+
+    def test_build_soc(self):
+        """Tests functionality of the build_soc method."""
+        in_kwargs = {'abcd': 'efgh'}
+        soc_builder = self.simple_init().build_soc(self.soc, **in_kwargs)
+        build_kwargs = soc_builder.build.call_args.kwargs
+
+        self.builder_constructor.assert_called_once()
+        soc_builder.build.assert_called_once()
+        self.assertEqual(soc_builder, self.builder)
+        self.assertEqual(in_kwargs['abcd'], build_kwargs['abcd'])
```

Comment:
This test could also check that whatever soc_core_argdict does is applied here, and then the above test wouldn't be needed :)